pkgs/tools/misc/qarma/default.nix

```diff
@@ -0,0 +1,39 @@
+{ mkDerivation
+, lib
+, fetchFromGitHub
+, wrapQtAppsHook
+, qtbase
+, qmake
+, qtx11extras ? null # qt5
+, qt5compat ? null # qt6
```

Comment:
We do not want to use ``? null`` in inputs.

Reply:
this is for compatibility with 

libsForQt5.callPackage
libsForQt6.callPackage

better solution?

Reply:
You can just drop the `? null` since you are using `lib.optionals` below and nix has lazy evaluation.

Reply:
> You can just drop the `? null`

nope:

> $ nix-build ./nixpkgs/ -A qarma-qt6
> error: anonymous function at /tmp/milahu/nixpkgs-qt6-init/nixpkgs/pkgs/tools/misc/qarma/default.nix:1:1 called without required argument 'qtx11extras'

and

> $ nix-build ./nixpkgs/ -A qarma
> error: anonymous function at /tmp/milahu/nixpkgs-qt6-init/nixpkgs/pkgs/tools/misc/qarma/default.nix:1:1 called without required argument 'qt5compat'